query/src/main/scala/filodb/query/exec/rangefn/RangeFunction.scala

```diff
@@ -339,12 +337,21 @@ object RangeFunction {
     }
   }
 
-  def histChunkedFunction(func: Option[InternalRangeFunction],
-                          funcParams: Seq[Any] = Nil,
-                          maxCol: Option[Int] = None): RangeFunctionGenerator = func match {
-    case None if maxCol.isDefined => () => new LastSampleChunkedFunctionHMax(maxCol.get)
+  def histMaxRangeFunction(f: Option[InternalRangeFunction]): Option[InternalRangeFunction] =
+    f match {
+      case None                   => Some(LastSampleHistMax)
+      case Some(SumOverTime)      => Some(SumAndMaxOverTime)
+      case other                  => other
+    }
+
+  def histChunkedFunction(schema: ResultSchema,
+                          func: Option[InternalRangeFunction],
+                          funcParams: Seq[Any] = Nil): RangeFunctionGenerator = func match {
     case None                 => () => new LastSampleChunkedFunctionH
-    case Some(SumOverTime) if maxCol.isDefined => () => new SumAndMaxOverTimeFuncHD(maxCol.get)
+    case Some(LastSampleHistMax) => require(schema.columns(2).name == "max")
+                                 () => new LastSampleChunkedFunctionHMax(schema.colIDs(2))
+    case Some(SumAndMaxOverTime) => require(schema.columns(2).name == "max")
+                                 () => new SumAndMaxOverTimeFuncHD(schema.colIDs(2))
```

Comment:
The max column discovery is earlier.... in `MultiSchemaPartitionsExec`.  By the time we pick `LastSampleHistMax` and `SumAndMaxOverTime` functions, it is already verified that the max column exists and would be in column 2.   This is part of the move to make discovery earlier rather than when picking functions.  (That's why its a require)